ext/opentelemetry-ext-otcollector/src/opentelemetry/ext/otcollector/metrics_exporter/__init__.py

```diff
@@ -0,0 +1,172 @@
+    MetricRecord,
+    MetricsExporter,
+    MetricsExportResult,
+    aggregate,
+)
+
+DEFAULT_ENDPOINT = "localhost:55678"
+
+logger = logging.getLogger(__name__)
+
+
+# pylint: disable=no-member
+class CollectorMetricsExporter(MetricsExporter):
+    """OpenTelemetry Collector metrics exporter.
+
+    Args:
+        endpoint: OpenTelemetry Collector OpenCensus receiver endpoint.
+        service_name: Name of Collector service.
+        host_name: Host name.
+        client: MetricsService client stub.
+    """
+
+    def __init__(
+        self,
+        endpoint=DEFAULT_ENDPOINT,
+        service_name=None,
+        host_name=None,
+        client=None,
+    ):
+        self.endpoint = endpoint
+        if client is None:
+            self.channel = grpc.insecure_channel(self.endpoint)
+            self.client = metrics_service_pb2_grpc.MetricsServiceStub(
+                channel=self.channel
+            )
+        else:
+            self.client = client
+
+        self.node = utils.get_node(service_name, host_name)
+
+    def export(
+        self, metric_records: Sequence[MetricRecord]
+    ) -> MetricsExportResult:
+        try:
+            responses = self.client.Export(
+                self.generate_metrics_requests(metric_records)
+            )
+
+            # Read response
+            for _ in responses:
+                pass
+
+        except grpc.RpcError:
+            return MetricsExportResult.FAILED_RETRYABLE
+
+        return MetricsExportResult.SUCCESS
+
+    def shutdown(self) -> None:
+        pass
+
+    def generate_metrics_requests(self, metrics):
+        collector_metrics = translate_to_collector(metrics)
+        service_request = metrics_service_pb2.ExportMetricsServiceRequest(
+            node=self.node, metrics=collector_metrics
+        )
+        yield service_request
+
+
+# pylint: disable=too-many-branches
+def translate_to_collector(metric_records: Sequence[MetricRecord]):
+    collector_metrics = []
+    for metric_record in metric_records:
+
+        label_values = []
+        label_keys = []
+        for label_tuple in metric_record.label_set.labels:
+            label_keys.append(metrics_pb2.LabelKey(key=label_tuple[0]))
+            label_values.append(
+                metrics_pb2.LabelValue(
+                    has_value=label_tuple[1] is not None, value=label_tuple[1]
+                )
+            )
+
+        metric_descriptor = metrics_pb2.MetricDescriptor(
+            name=metric_record.metric.name,
+            description=metric_record.metric.description,
+            unit=metric_record.metric.unit,
+            type=get_collector_metric_type(metric_record.metric),
+            label_keys=label_keys,
+        )
+
+        timeseries = metrics_pb2.TimeSeries(
+            start_timestamp=utils.proto_timestamp_from_time_ns(
+                metric_record.metric.get_handle(
+                    metric_record.label_set
+                ).last_update_timestamp
```

Comment:
Looks like there is not such thing in OpenTelemetry as the time when the metric was reset, not sending anything and leaving Collector to handle it would be a better approach here